src/modules/schulconnex/persistence/schulconnex.repo.integration.spec.ts

```diff
@@ -163,6 +167,143 @@ describe('Schulconnex Repo', () => {
+        });
+
+        it('should return all persons at all organisation with correct serviceProvider-Role', async () => {
+            const personA: Person<true> = await createPerson();
+            const personB: Person<true> = await createPerson();
+            const personC: Person<true> = await createPerson();
+            const serviceProviderA: ServiceProvider<true> = await serviceProviderRepo.save(
+                DoFactory.createServiceProvider(false),
+            );
+            const rolleA: Rolle<true> | DomainError = await rolleRepo.save(
+                DoFactory.createRolle(false, { serviceProviderIds: [serviceProviderA.id] }),
+            );
+            const rolleB: Rolle<true> | DomainError = await rolleRepo.save(DoFactory.createRolle(false));
+            const organisationA: Organisation<true> = await organisationRepository.save(
+                DoFactory.createOrganisation(false),
+            );
+            const organisationB: Organisation<true> = await organisationRepository.save(
+                DoFactory.createOrganisation(false),
+            );
+            if (rolleA instanceof DomainError) throw Error();
+            if (rolleB instanceof DomainError) throw Error();
+
+            await rollenErweiterungRepo.create(
+                DoFactory.createRollenerweiterung(false, {
+                    rolleId: rolleA.id,
+                    organisationId: organisationA.id,
+                    serviceProviderId: serviceProviderA.id,
+                }),
+            );
+
+            await Promise.all([
+                personenkontextRepoInternal.save(
+                    createPersonenkontext(false, {
+                        personId: personA.id,
+                        rolleId: rolleA.id,
+                        organisationId: organisationA.id,
+                    }),
+                ),
+                personenkontextRepoInternal.save(
+                    createPersonenkontext(false, {
+                        personId: personB.id,
+                        rolleId: rolleA.id,
+                        organisationId: organisationB.id,
+                    }),
+                ),
+                personenkontextRepoInternal.save(
+                    createPersonenkontext(false, {
+                        personId: personC.id,
+                        rolleId: rolleB.id,
+                        organisationId: organisationA.id,
+                    }),
+                ),
+            ]);
+
+            const personIds: PersonID[] =
+                await sut.findPersonIdsWithRollenerweiterungForServiceProviderAndOptionallyOrganisations(
+                    new Set([serviceProviderA.id]),
+                    'all',
+                );
+            expect(personIds).toHaveLength(2);
+            expect(personIds.findIndex((id: PersonID) => id === personA.id)).not.toEqual(-1);
+            expect(personIds.findIndex((id: PersonID) => id === personB.id)).not.toEqual(-1);
```

Comment:
This should also work and is a bit easier to read imo
```suggestion
            expect(personIds).toContain(personA.id);
            expect(personIds).toContain(personB.id);
```